packages/design-system/src/components/Dialog/Dialog.tsx

```diff
@@ -334,86 +148,90 @@ export const Dialog = (props: DialogProps) => {
     };
   }, []);
 
-  const {
-    actions,
-    actionsClassName,
-    analytics,
-    ariaCloseLabel,
-    children,
-    className,
-    closeButtonSize,
-    closeButtonText,
-    closeButtonVariation,
-    closeIcon,
-    closeText,
-    escapeExits,
-    escapeExitDisabled,
-    headerClassName,
-    heading,
-    onExit,
-    size,
-    title,
-    ...modalProps
-  } = props;
+  // Toggle dialog open
+  const dialogRef = useRef(null);
+  const lastActiveElement = React.useRef(null);
+  const firstRender = React.useRef(true);
+
+  useLayoutEffect(() => {
+    if (window.HTMLDialogElement === undefined) {
+      dialogPolyfill.registerDialog(dialogRef.current);
+      // import('dialog-polyfill/dist/dialog-polyfill.css');
+    }
+  }, []);
+
+  useLayoutEffect(() => {
+    // prevents calling imperative methods on mount since the polyfill will throw an error since we are not using the `open` attribute
+    if (firstRender.current) {
+      firstRender.current = false;
+    } else {
+      const dialogNode = dialogRef.current;
+      if (open && type === 'modal') {
+        lastActiveElement.current = document.activeElement;
+        dialogNode.showModal();
+      } else if (open && type !== 'modal') {
+        lastActiveElement.current = document.activeElement;
+        dialogNode.show();
+      } else {
+        dialogNode.close();
+        lastActiveElement.current.focus();
+      }
+    }
+  }, [open, type]);
+
+  // PE: Implement `onCancel={onExit}` for `dialog`
+  // https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/oncancel
+  // onCancel allows close/open of dialog using keyboard - state doesn't get tripped up
+  // Receiving TS error saying `onCancel` doesn't exist on HTMLDialogElement
+  useEffect(() => {
+    const dialogNode = dialogRef.current;
+    const handleCancel = (event) => {
+      event.preventDefault();
+      onExit();
+    };
+    dialogNode.addEventListener('cancel', handleCancel);
```

Comment:
TS kept yelling at me for adding `onCancel` to the dialog, so I'm adding it here